raiden/raiden_service.py

```diff
@@ -134,6 +136,20 @@ def save_snapshot(serialization_file, raiden):
         )
 
 
+def endpoint_registry_exception_handler(greenlet):
+    try:
+        greenlet.get()
+    except (JSONRPCPollTimeoutException, Exception) as e:
+        if (e.message == "timeout when polling for transaction" or
+           isinstance(e, JSONRPCPollTimeoutException)):
+                log.fatal("Endpoint registry failed: %s. "
+                          "Ethereum RPC API might be unreachable.",
+                          repr(e))
+        else:
+            log.fatal("Endpoint registry failed: %s. ", repr(e))
```

Comment:
Traceback printed by gevent is much more useful; if you use exc_info, it'll merely print a traceback of the exception handler. See below, former is gevent's log.

`
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/tmp/popo.py", line 27, in _run
    raise Exception('ooops')
Exception: ooops
<MyGreenlet at 0x7f1503ce5410> failed with Exception

ERROR:__main__:XXXXX
Traceback (most recent call last):
  File "/tmp/popo.py", line 12, in on_exception
    greenlet.get()
  File "/usr/lib64/python2.7/site-packages/gevent/greenlet.py", line 251, in get
    raise self._exception
Exception: ooops
`